jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/CaseServiceImplTest.java

```diff
@@ -3699,14 +3705,31 @@ public void testCaseWithIndexedFromVariableUpdatesResetVariable() {
             }
         }
     }
-    
+
+    @Before
+    public void setUp() throws Exception {
+        registerListenerMvelDefinition("org.jbpm.casemgmt.impl.util.CountDownListenerFactory.get(\"" +
+                                       getClass().getSimpleName() + "1\", " +
+                                       "\"end 1\", 1)");
+        registerListenerMvelDefinition("org.jbpm.casemgmt.impl.util.CountDownListenerFactory.get(\"" +
+                                       getClass().getSimpleName() + "2\", " +
+                                       "\"Stage 1\", 1)");
```

Comment:
For better test performance, perhaps this can be done overriding getProcessListeners, and registering based on test method name. This was my proposal (overlapped with yours):
https://github.com/elguardian/jbpm/pull/4/commits/461ce872c916a3c0ab81b3cb80a7b3ef41e8e9e8

Reply:
that solution is a bit more confusing and error prone.